File: Net/01/TasksManager.Application/Repositories/ToDoTasksRepository.cs
using Cqrs;
using System;
using System.Collections.Concurrent;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TasksManager.Repositories
{
    public class ToDoTasksRepository : IRepository<ToDoTaskDao>
    {
        private static ConcurrentDictionary<Guid, ToDoTaskDao> _storage = new ConcurrentDictionary<Guid, ToDoTaskDao>();

        public void Delete(Guid id)
        {
            ToDoTaskDao deleted;
            if (_storage.ContainsKey(id)) _storage.TryRemove(id, out deleted);
        }

        public List<ToDoTaskDao> GetAll()
        {
            return _storage.Values.ToList();
        }

        public ToDoTaskDao GetById(Guid id)
        {
            if (_storage.ContainsKey(id)) return _storage[id];
            return null;
        }

        public ToDoTaskDao Save(ToDoTaskDao item)
        {
            _storage[item.Id] = item;
            return item;
        }
    }
}

File: Java/01/tasks-manager-application/src/main/java/org/tasksmanager/Repositories/ToDoTasksRepository.java
package org.tasksmanager.Repositories;

import java.util.List;
import java.util.UUID;
import java.util.concurrent.ConcurrentHashMap;
import org.cqrs.Repository;


    public class ToDoTasksRepository implements Repository<ToDoTaskDao>
    {
        private static ConcurrentHashMap<UUID, ToDoTaskDao> _storage = new ConcurrentHashMap<UUID, ToDoTaskDao>();

        public void Delete(UUID id)
        {
            if (_storage.containsKey(id)) _storage.remove(id);
        }

        public List<ToDoTaskDao> GetAll()
        {
            return (List<ToDoTaskDao>) _storage.elements();
        }

        public ToDoTaskDao GetById(UUID id)
        {
            if (_storage.containsKey(id)) return _storage.get(id);
            return null;
        }

        public ToDoTaskDao Save(ToDoTaskDao item)
        {
            _storage.put(item.getId(), item);
            return item;
        }
    }

